nix/cardano-graphql.nix

```diff
@@ -0,0 +1,75 @@
+{ mkYarnPackage
+, lib
+, cardano-graphql-src
+, runtimeShell
+, nodejs
+, python
+}:
+
+let
+  packageJSON = cardano-graphql-src + "/package.json";
+  version = (__fromJSON (__readFile packageJSON)).version;
+
+in mkYarnPackage {
+  pname = "cardano-graphql";
+  inherit packageJSON version;
+  yarnLock = cardano-graphql-src + "/yarn.lock";
+  src = lib.cleanSourceWith {
+    filter = name: type: let
+      baseName = baseNameOf (toString name);
+      sansPrefix = lib.removePrefix (toString ../.) name;
+      in_blacklist =
+        lib.hasPrefix "/node_modules" sansPrefix ||
+        lib.hasPrefix "/build" sansPrefix ||
+        lib.hasPrefix "/.git" sansPrefix;
+      in_whitelist =
+        (type == "directory") ||
+        (lib.hasSuffix ".yml" name) ||
+        (lib.hasSuffix ".ts" name) ||
+        (lib.hasSuffix ".json" name) ||
+        (lib.hasSuffix ".graphql" name) ||
+        baseName == "package.json" ||
+        baseName == "yarn.lock" ||
+        (lib.hasPrefix "/deploy" sansPrefix);
+    in (
+      (!in_blacklist) && in_whitelist
+    );
+    src = cardano-graphql-src;
+  };
+  yarnPreBuild = ''
+    mkdir -p $HOME/.node-gyp/${nodejs.version}
+    echo 9 > $HOME/.node-gyp/${nodejs.version}/installVersion
+    ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
+  '';
+  pkgConfig = {
+    node-sass = {
+      buildInputs = [ python ];
+      postInstall = ''
+        yarn --offline run build
+      '';
+    };
+  };
+
+  installPhase = ''
+    unpackPhase
+    cd $sourceRoot
```

Comment:
why the `unpackPhase` here? it should already have been unpacked by previous phases